src/components/Swap/Input.tsx

```diff
@@ -52,6 +52,30 @@ export const InputColumn = styled(Column)<{ approved?: boolean; hasColor?: boole
     filter: ${({ approved }) => (approved ? undefined : 'saturate(0) opacity(0.4)')};
     transition: filter 0.25s;
   }
+
+  &:before {
+    box-sizing: border-box;
+    background-size: 100%;
+    border-radius: inherit;
+
+    position: absolute;
+    top: 0;
+    left: 0;
+
+    width: 100%;
+    height: 100%;
+    pointer-events: none;
+    content: '';
+    border: 1px solid ${({ theme }) => theme.module};
```

Comment:
Why can't you just add `border` to the `InputColumn`?

Reply:
adding a border changes the size of the element so the UI jumps a bit, whereas the pseudo overlays. i had the same thought 😄